planning/behavior_path_planner/src/utils/lane_change/utils.cpp

```diff
@@ -1103,4 +1103,16 @@ ExtendedPredictedObject transform(
 
   return extended_object;
 }
+
+bool isCollidedPolygonsInLanelet(
+  const std::vector<Polygon2d> & collided_polygons, const lanelet::ConstLanelets & lanes)
+{
+  const auto lanes_polygon = createPolygon(lanes, 0.0, std::numeric_limits<double>::max());
+
+  const auto is_in_lanes = [&](const auto & collided_polygon) {
+    return lanes_polygon && boost::geometry::intersects(lanes_polygon.value(), collided_polygon);
```

Comment:
@zulfaqar-azmi-t4 
[NITS]
You can use this function for fast cumputation (it's not necessary to do it now though)
https://github.com/autowarefoundation/autoware.universe/blob/16dc5358ee9ac3934190c0f09fb9d782500cb91d/common/tier4_autoware_utils/src/geometry/geometry.cpp#L362

Reply:
I'll make a separate PR in the future. 👍🏼